Earthfile

```
@@ -25,6 +25,7 @@ lint:
 
 # build builds Crossplane for your native OS and architecture.
 build:
+  BUILD +go-build-locally
```

```suggestion
  ARG USERPLATFORM
  BUILD +go-build --platform=${USERPLATFORM}
```

Would this achieve the goal of causing Earthly to build binaries for the user's platform (e.g. Darwin)?

It works for me, but I run Earthly on Linux so for me USERPLATFORM == NATIVEPLATFORM.

---

I believe I have tested this suggestion out on my Mac M1 but it didn't seem to work for me. I synced to latest from https://github.com/tampakrap/crossplane/tree/earthfile_build_step_also_build_go, applied this suggestion on top, and ran `earthly +build`.

I do see build output that looks correct: `+go-build | platform=darwin/arm64`, but the `_output` dir still only has `_output/bin/linux_arm64/`.

A few more details about my testing are available in the following gist: 
https://gist.github.com/jbw976/d7a3d230b5cfaefb4259f50dbe056c4a

---

aha, if I just pass the `--platform` arg before the `+go-build` target, then it does work as intended! e.g.:

```
BUILD --platform=${USERPLATFORM} +go-build
```

Gives me:
```
❯ tree _output
_output
├── bin
│   └── darwin_arm64
│       ├── crank
│       └── crossplane
└── charts
    └── crossplane-0.0.0-1717748817-e9886233.tgz

3 directories, 3 files
```

---

@negz you are absolutely right! I tried the following and I could swear that it showed linux in all cases, but now I tried it again and indeed USERPLATFORM shows the correct one! Many thanks for spotting this and ultimately providing the correct solution!

```Dockerfile
# theo is testing stuff
theo:
  ARG NATIVEPLATFORM
  ARG USERPLATFORM
  ARG TARGETPLATFORM
  FROM nginx:alpine
  RUN echo "NATIVEPLATFORM is $NATIVEPLATFORM"
  RUN echo "USERPLATFORM is $USERPLATFORM"
  RUN echo "TARGETPLATFORM is $TARGETPLATFORM"
```

```
               +theo | --> RUN echo "NATIVEPLATFORM is $NATIVEPLATFORM"
               +theo | NATIVEPLATFORM is linux/amd64
               +theo | --> RUN echo "USERPLATFORM is $USERPLATFORM"
               +theo | USERPLATFORM is darwin/amd64
               +theo | --> RUN echo "TARGETPLATFORM is $TARGETPLATFORM"
               +theo | TARGETPLATFORM is linux/amd64
```